test/lib/notifier_test.rb

```diff
@@ -58,14 +85,29 @@ def test_bucket_path
   end
 
   def test_object_key
+    stub_client = Aws::S3::Client.new(stub_responses: true)
+    stubbed_objs = {
+      contents: [
+        { key: 'DONE.txt' },
+      ]
+    }
+    stub_client.stub_responses(:list_objects_v2, stubbed_objs)
```

Comment:
if you don't want to repeat this scaffolding you can move it into the setup method like
```ruby
class MyTest
  def setup
    @stub_client = Aws::S3::Client.new(stub_responses: true)
        stubbed_objs = {
      contents: [
        { key: 'DONE.txt' },
      ]
    }
    @stub_client.stub_responses(:list_objects_v2, stubbed_objs)
  end
```
and then the new `@stub_client` instance var wherever you need it.